org.adempiere.base.process/src/org/compiere/process/CopyRole.java

```diff
@@ -43,7 +44,7 @@ public class CopyRole extends SvrProcess
 {
 	private int m_AD_Role_ID_From = 0;
 	private int m_AD_Role_ID_To = 0;
-	private int m_AD_Client_ID = 0;	
+	private Integer m_AD_Client_ID = null;	

```

Comment:
Since the process has no AD_Client_ID parameter, this should be removed.

Reply:
The name.equals("AD_Client_ID") part in prepare() should be removed too.

Reply:
This validation was done because when the modified AD_Client parameter is not filled in in the process, it takes the logged-in tenant and does not put zero in the AD_Client field

Reply:
This modification from int to integer was made to validate whether the AD_Client parameter was filled in or not, as it is not mandatory, in case the AD_Client is not informed in the parameter we get  the AD_Client of the context

Reply:
Changing it to integer we can assign null to the variable, enabling the validation if it was filled or not, since when assigned 0 when initializing it, we would not be sure if it was filled, since the dictionary has ID 0

Reply:
No, I means since the process actually doesn't have an "AD_Client_ID" parameter, the copy role process class shouldn't have the m_AD_Client_ID instance variable at all. You just always get it from the context through Env.getAD_Client_ID call.

Reply:
Yes, I understand the case, but the copyRole process in the dictionary has the AD_client parameter, which was one of the parameters that I add dynamic validation

Reply:
ok, my mistake. checking with Carlos in ticket why do we need to have the client/tenant process parameter ? looks a strange one to me.

Reply:
Adjustments made

Tenant will be obtained through context

Validation created for only Roles that the user has access removed

Adjustments to the naming of the Role_ID_From and Role_ID_To parameters

Improve the description of the process to make it clear that this will be destructive to previous Role to dat